deepchem/models/torch_models/grover.py

```diff
@@ -92,3 +98,312 @@ def forward(self, graph_batch: List[torch.Tensor]):
+                sigmoid(preds["fg_task"]["bond_from_atom"]),
+                sigmoid(preds["fg_task"]["bond_from_bond"]))
+
+        av_loss = av_atom_loss + av_bond_loss
+        bv_loss = bv_atom_loss + bv_bond_loss
+        fg_atom_loss = fg_atom_from_atom_loss + fg_atom_from_bond_loss
+        fg_bond_loss = fg_bond_from_atom_loss + fg_bond_from_bond_loss
+
+        fg_loss = fg_atom_loss + fg_bond_loss
+        fg_dist_loss = fg_atom_dist_loss + fg_bond_dist_loss
+
+        overall_loss = av_loss + bv_loss + fg_loss + dist_coff * av_dist_loss + \
+                       dist_coff * bv_dist_loss + fg_dist_loss
+
+        return overall_loss
+
+    @staticmethod
+    def atom_random_mask(atom_vocab, smiles_batch: List[str]):
+        """
+        Parameters
+        ----------
+        atom_vocab: grover.data.MolVocab
+            atom vocabulary
+        smiles_batch: List[str]
+            a list of smiles string
+        """
+        # This is a zero padding.
+        vocab_label = [0]
+        percent = 0.15
+        for smi in smiles_batch:
+            mol = Chem.MolFromSmiles(smi)
+            mlabel = [0] * mol.GetNumAtoms()
+            n_mask = math.ceil(mol.GetNumAtoms() * percent)
+            perm = np.random.permutation(mol.GetNumAtoms())[:n_mask]
+            for p in perm:
+                atom = mol.GetAtomWithIdx(int(p))
+                mlabel[p] = atom_vocab.stoi.get(atom_to_vocab(mol, atom),
+                                                atom_vocab.other_index)
+
+            vocab_label.extend(mlabel)
+        return vocab_label
+
+    @staticmethod
+    def bond_random_mask(bond_vocab, smiles_batch):
+        """
+        Parameters
+        ----------
+        bond_vocab: MolVocab
+            bond vocabulary
+        smiles_batch: List[str]
+            List of smiles strings
+        """
+        # This is a zero padding.
+        vocab_label = [0]
+        percent = 0.15
+        for smi in smiles_batch:
+            mol = Chem.MolFromSmiles(smi)
+            nm_atoms = mol.GetNumAtoms()
+            nm_bonds = mol.GetNumBonds()
+            mlabel = []
+            n_mask = math.ceil(nm_bonds * percent)
+            perm = np.random.permutation(nm_bonds)[:n_mask]
+            virtual_bond_id = 0
+            for a1 in range(nm_atoms):
+                for a2 in range(a1 + 1, nm_atoms):
+                    bond = mol.GetBondBetweenAtoms(a1, a2)
+
+                    if bond is None:
+                        continue
+                    if virtual_bond_id in perm:
+                        label = bond_vocab.stoi.get(bond_to_vocab(mol, bond),
+                                                    bond_vocab.other_index)
+                        mlabel.extend([label])
+                    else:
+                        mlabel.extend([0])
+
+                    virtual_bond_id += 1
+            vocab_label.extend(mlabel)
+        return vocab_label
+
+
+class GroverFinetuneModel(TorchModel):
```

Comment:
The convention we are following in `InfoGraphModel` is to have the same model use a flag to swap between pretraining/fine-tuning. We should follow the same convention here